elasticsearch/src/test/java/org/hibernate/search/elasticsearch/test/GsonStreamedEncodingTest.java

```diff
@@ -101,6 +101,7 @@ public void testContentIsRepeatable() {
 		list.add( buildEmptyJSON() );
 		try ( GsonHttpEntity entity = new GsonHttpEntity( gson, list ) ) {
 			final byte[] productionOne = produceContentWithCustomEncoder( entity );
+			entity.close();
```

Comment:
I'm confused about this commit. The `entity` is in an auto-closeable block so it should get closed?
Also, why close it here when it's still used on the next line?
Finally, if I revert this nothing seems to fail.

Reply:
We are **re-using** the entity on the next line, expecting to "rewind". So we need to send a message to tell it to rewing, and this message is the `close()` method. We actually can re-use it after calling `close()` because that's the definition of being "repeatable".
I know it works without closing it, it's just that it's not required that it does... So why enforce it? Besides, we also test for "repeatability" further in the same file, and *there* we close the entity before re-using it.

Reply:
great point! Thanks, sorry that makes perfect sense I probably reviewed this too late in the evening